p2p/src/peers/mod.rs

```diff
@@ -158,6 +158,16 @@ impl Peers {
             .collect()
     }
 
+    /// Remove a peer given an index from new addresses bucket
+    /// Returns the removed addresses
+    pub fn remove_from_new_with_index(&mut self, indexes: &[u16]) -> Vec<SocketAddr> {
```

Comment:
Are we certain this will always succeed and not remove the wrong entry?

Reply:
Due that new bucket needs the information about the source address, it is difficult to remove without that information. So using the index we can remove directly from the hashmap.
The other solution could be to have 2 structs for PeerInfo, one without source for the tried bucket and another with source for the new bucket